src/interface/mesh_data.hpp

```diff
@@ -242,8 +242,24 @@ class MeshData {
 
   void Set(BlockList_t blocks, Mesh *pmesh, int ndim);
   void Set(BlockList_t blocks, Mesh *pmesh);
-  void Initialize(const MeshData<T> *src, const std::vector<std::string> &names,
-                  const bool shallow);
+
+  template <typename ID_t>
+  void Initialize(const MeshData<T> *src, const std::vector<ID_t> &vars,
+                  const bool shallow) {
+    if (src == nullptr) {
+      PARTHENON_THROW("src points at null");
+    }
+    pmy_mesh_ = src->GetParentPointer();
+    const int nblocks = src->NumBlocks();
+    block_data_.resize(nblocks);
+
+    grid = src->grid;
+    for (int i = 0; i < nblocks; ++i) {
+      block_data_[i] =
+          GetSourceBlockPointer_(i, src, pmy_mesh_)
+              ->meshblock_data.Add(stage_name_, src->GetBlockData(i), vars, shallow);
+    }
+  }
```

Comment:
Maybe I am missing something, but this looks like it will just grab the first `nblocks` blocks in the meshblock list for a `Mesh` on a given rank. Doesn't this mean the wrong blocks will be grabbed if there are multiple partitions?

Reply:
Discussed offline. You're right---this is a bug. Fixed, and added a comment discussing an edge case we need to be concerned about. But not fixing it right now.